airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -447,44 +479,64 @@ def generate_id_hashing_model(self, from_table: str, column_names: Dict[str, Tup
         return sql
 
     def safe_cast_to_strings(self, column_names: Dict[str, Tuple[str, str]]) -> List[str]:
+
         return [StreamProcessor.safe_cast_to_string(self.properties[field], column_names[field][1]) for field in column_names]
 
     @staticmethod
     def safe_cast_to_string(definition: Dict, column_name: str) -> str:
         """
-        Note that the result from this static method should always be used within a jinja context (for example, from jinja macro surrogate_key call)
+        Note that the result from this static method should always be used within a
+        jinja context (for example, from jinja macro surrogate_key call)
+
+        The jinja_remove function is necessary because of Oracle database, some columns
+        are created with {{ quote('column_name') }} and reused the same fields for this
+        operation. Because the quote is injected inside a jinja macro we need to remove
+        the curly brackets.
         """
+        quote_in_parenthesis = re.compile(r"quote\((.*)\)")
         if "type" not in definition:
-            return column_name
+            col = column_name
         elif is_boolean(definition["type"]):
-            return f"boolean_to_string({column_name})"
+            col = f"boolean_to_string({column_name})"
         elif is_array(definition["type"]):
-            return f"array_to_string({column_name})"
+            col = f"array_to_string({column_name})"
         else:
-            return column_name
+            col = column_name
+
+        if quote_in_parenthesis.findall(col):
+            return remove_jinja(col)
```

Comment:
should we restrict this to only ORACLE to denote that this is a concern for that destination only?